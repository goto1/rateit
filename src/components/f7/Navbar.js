import React from "react";
import PropTypes from "prop-types";

export const NavLeft = ({ backLink, sliding = false, className }) => {
  const classNames = `left ${sliding ? "sliding" : ""} ${className
    ? className
    : ""} `.trim();
  return (
    <div className={classNames}>
      <a className="back link">
        <i className="icon icon-back" />
        <span>
          {backLink}
        </span>
      </a>
    </div>
  );
};

NavLeft.propTypes = {
  backLink: PropTypes.string.isRequired,
  sliding: PropTypes.bool,
  className: PropTypes.string
};

export const NavCenter = ({ sliding = false, className, children }) => {
  const classNames = `center ${sliding ? "sliding" : ""} ${className
    ? className
    : ""}`.trim();
  return (
    <div className={classNames}>
      {children}
    </div>
  );
};

NavCenter.propTypes = {
  sliding: PropTypes.bool,
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export const NavRight = ({ sliding = false, className, children }) => {
  const classNames = `right ${sliding ? "sliding" : ""} ${className
    ? className
    : ""}`.trim();
  return (
    <div className={classNames}>
      {children}
    </div>
  );
};

NavRight.propTypes = {
  sliding: PropTypes.bool,
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

const Navbar = ({ className, children }) => {
  const classNames = `navbar ${className ? className : ""}`.trim();
  return (
    <div className={classNames}>
      <div className="navbar-inner">
        {children}
      </div>
    </div>
  );
};

Navbar.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default Navbar;
