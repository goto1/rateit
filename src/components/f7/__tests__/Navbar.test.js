import React from "react";
import Navbar, { NavLeft, NavCenter, NavRight } from "../Navbar";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("Navbar component", () => {
  it("should render correctly", () => {
    const tree = renderer.create(
      <Navbar>
        <NavLeft backLink="Back" />
        <NavCenter>Home</NavCenter>
        <NavRight>See More</NavRight>
      </Navbar>
    );
    expect(tree).toMatchSnapshot();
  });

  it("should render NavLeft component", () => {
    const wrapper = shallow(
      <Navbar>
        <NavLeft backLink="Back" />
      </Navbar>
    );
    const actual = wrapper.find(NavLeft).length;
    const expected = 1;
    expect(actual).toBe(expected);
  });

  it("should render NavCenter component", () => {
    const wrapper = shallow(
      <Navbar>
        <NavCenter>Home</NavCenter>
      </Navbar>
    );
    const actual = wrapper.find(NavCenter).length;
    const expected = 1;
    expect(actual).toBe(expected);
  });

  it("should render NavRight component", () => {
    const wrapper = shallow(
      <Navbar>
        <NavRight>See more</NavRight>
      </Navbar>
    );
    const actual = wrapper.find(NavRight).length;
    const expected = 1;
    expect(actual).toBe(expected);
  });

  describe("NavLeft", () => {
    it("should render as a sliding element", () => {
      const wrapper = shallow(<NavLeft backLink="Back" sliding={true} />);
      const actual = wrapper.props().className.includes("sliding");
      const expected = true;
      expect(actual).toBe(expected);
    });

    it("should render the back link text", () => {
      const wrapper = shallow(<NavLeft backLink="Back" />);
      const actual = wrapper.find("span").at(0).html().includes("Back");
      const expected = true;
      expect(actual).toBe(expected);
    });
  });

  describe("NavCenter", () => {
    it("should render with a custom styling class", () => {
      const wrapper = shallow(<NavCenter className="custom">Home</NavCenter>);
      const actual = wrapper.props().className.includes("custom");
      const expected = true;
      expect(actual).toBe(expected);
    });
  });

  describe("NavRight", () => {
    it("should render a span element", () => {
      const wrapper = shallow(
        <NavRight>
          <span>Hello</span>
        </NavRight>
      );
      const actual = wrapper.find("span").length;
      const expected = 1;
      expect(actual).toBe(expected);
    });
  });
});
