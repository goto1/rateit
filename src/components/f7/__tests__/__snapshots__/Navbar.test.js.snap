// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render correctly 1`] = `


<div
  className="navbar"
>
  <div
    className="navbar-inner"
  >
    <div
      className="left"
    >
      <a
        className="back link"
      >
        <i
          className="icon icon-back"
        />
        <span>
          Back
        </span>
      </a>
    </div>
    <div
      className="center"
    >
      Home
    </div>
    <div
      className="right"
    >
      See More
    </div>
  </div>
</div>
`;
